src/Helper/PaymentHelper.php

```diff
@@ -52,13 +53,15 @@ public function webformSubmissionPresave(WebFormSubmissionInterface $submission)
      * as a JSON object in the os2forms_payment submission value.
      */
     $request = $this->requestStack->getCurrentRequest();
-    $payment_reference_field = $request->request->get('payment_reference_field');
+    $payment_reference_field = $request->request->get('os2forms_payment_reference_field');
+    $paymentPosting = $payment_element['#payment_posting'] ?? 'unset';
 
     if ($request && $amount_to_pay) {
       $payment_object = [
         'paymentObject' => [
           'payment_id' => $payment_reference_field,
           'amount' => $amount_to_pay,
+          'posting' => $paymentPosting,
```

Comment:
Be consistent in how you format variable names. Use camelCase. For all variables in your classes.